### DISCUSSION TITLE
Ignoring .js file only if corresponding .ts file present

### DISCUSSION THREAD

**Discussion by manikantag** [BurntSushi/ripgrep#2465] [Q&A] [answered]
Hi,

Is there a way to ignore generated `.js` files only if a corresponding `.ts` file exist in the same directory? Either using some glob pattern or custom type?

Example:

```
dir1/dir2/about.js
dir1/dir2/home.ts
dir1/dir2/index.js     # <----- ignore this as corresponding `.ts.` file exists (next file)
dir1/dir2/index.ts
```
Given above file structure, I want to ignore only `index.js` as corresponding `index.ts` file exists.

I use `--json` to process the results afterwards if that matters.

Thanks

**Comment by BurntSushi** [comment, marked as answer]
Nope. There is no way to hook into ripgrep's filtering logic and introduce a dependency based on whether some other file with a similar name exists.

My suggestion would be to add your js files that your want to ignore to an `.rgignore` file. It seems likely you could write a relatively simple program to walk your tree and produce those ignore files automatically.

**Comment by manikantag** [comment]
Thanks for confirming that I'm not missing any existing option. Yeah, my thought is to generate ignore file dynamically for every search or filter out during the UI creation time from the JSON results. Later is little trickier as `.js` file matches may come early than `.ts` file matches in the JSON results.

Thanks
